velox/expression/CastExpr-inl.h

```diff
@@ -453,6 +453,42 @@ void CastExpr::applyIntToDecimalCastKernel(
       });
 }
 
+template <typename TOutput>
+void CastExpr::applyDoubleToDecimalCastKernel(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& toType,
+    VectorPtr& result) {
+  const auto sourceVector = input.as<SimpleVector<double>>();
+  auto rawResults =
+      result->asUnchecked<FlatVector<TOutput>>()->mutableRawValues();
+  const auto toPrecisionScale = getDecimalPrecisionScale(*toType);
+
+  applyToSelectedNoThrowLocal(context, rows, result, [&](vector_size_t row) {
+    if (sourceVector->isNullAt(row)) {
```

Comment:
Will this ever be the case? Are we not handling nulls elsewhere?

Reply:
I checked and found this branch would not be entered. Removed this branch and I will open a PR to remove similar checks of other cast kernels.

Reply:
Perfect. Thanks.

Reply:
Opened https://github.com/facebookincubator/velox/pull/8557 for this minor fix. Thanks.